### PR TITLE
[Payum] Provide easier way to extend assert action

### PR DESCRIPTION
**Pull request by lchrusciel**
Proposal how can we design actions, so they are easier to extend

Follow up to #98

### CODE REVIEW

File: src/Payum/Action/AssertAction.php [hunk 1]
```diff
@@ -7,15 +7,20 @@
...
 use Sylius\Component\Core\Model\PaymentInterface;
 
 final class AssertAction implements ActionInterface
 {
-    public function __construct(private SaferpayClientInterface $saferpayClient)
-    {
+    public function __construct(
+        private SaferpayClientInterface $saferpayClient,
+        private SuccessfulResponseHandlerInterface $successfulResponseHandler,
+        private FailedResponseHandlerInterface $failedResponseHandler,
```
Comment (lchrusciel) [inline src/Payum/Action/AssertAction.php:22]: Does it make sense to introduce service locator based on response? /cc @Zales0123

Comment (Zales0123) [reply src/Payum/Action/AssertAction.php:22]: Maybe, but probably not needed for now for only 2 handlers 🖖

File: spec/Payum/Action/Assert/FailedResponseHandlerSpec.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+namespace spec\CommerceWeavers\SyliusSaferpayPlugin\Payum\Action\Assert;
+
+use CommerceWeavers\SyliusSaferpayPlugin\Client\ValueObject\ErrorResponse;
+use CommerceWeavers\SyliusSaferpayPlugin\Payum\Action\ErrorName;
+use CommerceWeavers\SyliusSaferpayPlugin\Payum\Action\StatusAction;
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\PaymentInterface;
+
+class FailedResponseHandlerSpec extends ObjectBehavior
```
Comment (Zales0123) [inline spec/Payum/Action/Assert/FailedResponseHandlerSpec.php:13]: `final` :dancer:

File: spec/Payum/Action/Assert/SuccessfulResponseHandlerSpec.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+namespace spec\CommerceWeavers\SyliusSaferpayPlugin\Payum\Action\Assert;
+
+use CommerceWeavers\SyliusSaferpayPlugin\Client\ValueObject\AssertResponse;
+use CommerceWeavers\SyliusSaferpayPlugin\Client\ValueObject\Body\PaymentMeans;
+use CommerceWeavers\SyliusSaferpayPlugin\Client\ValueObject\Body\Transaction;
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\PaymentInterface;
+
+class SuccessfulResponseHandlerSpec extends ObjectBehavior
```
Comment (Zales0123) [inline spec/Payum/Action/Assert/SuccessfulResponseHandlerSpec.php:13]: And here 🖖

**Review by Zales0123 — APPROVED** [review]

**Comment by Zales0123** [comment]
🫡

**Pull request merged**